Source/WebCore/platform/SharedBuffer.h

```diff
@@ -299,6 +301,8 @@ class SharedBuffer : public FragmentedSharedBuffer {
     WEBCORE_EXPORT GRefPtr<GBytes> createGBytes() const;
 #endif
 
+    Ref<FragmentedSharedBuffer> asFragmentedSharedBuffer() const { return const_cast<SharedBuffer&>(*this); }
```

Comment:
Can't this return a Ref<const FragmentedSharedBuffer> rather than introduce a const_cast?

Reply:
What do you mean by returning a Ref? Doesn’t it already return a Ref? I tried several things but haven’t been able to get rid of this const cast. I’d be happy to try more things though.

Reply:
If you meant `Ref<const WebCore::FragmentedSharedBuffer> asFragmentedSharedBuffer();`, I tried but the IPC decoder does like it:
```
/Volumes/Work/WebKit/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/ArgumentCoder.h:61:15: error: calling a protected constructor of class 'WebCore::FragmentedSharedBuffer'
            T t;
In file included from /Volumes/Work/WebKit/OpenSource/WebKitBuild/Debug/DerivedSources/WebKit/GeneratedSerializers.mm:27:
In file included from /Volumes/Work/WebKit/OpenSource/WebKitBuild/Debug/DerivedSources/WebKit/GeneratedWebKitSecureCoding.h:28:
In file included from /Volumes/Work/WebKit/OpenSource/Source/WebKit/Shared/Cocoa/CoreIPCTypes.h:28:
In file included from /Volumes/Work/WebKit/OpenSource/Source/WebKit/Shared/Cocoa/CoreIPCArray.h:30:
In file included from /Volumes/Work/WebKit/OpenSource/Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.h:28:
/Volumes/Work/WebKit/OpenSource/Source/WebKit/Platform/IPC/ArgumentCoders.h:309:34: note: in instantiation of function template specialization 'IPC::ArgumentCoder<const WebCore::FragmentedSharedBuffer>::decode<IPC::Decoder>' requested here
        return ArgumentCoder<U>::decode(decoder);
In file included from /Volumes/Work/WebKit/OpenSource/WebKitBuild/Debug/DerivedSources/WebKit/GeneratedSerializers.mm:27:
In file included from /Volumes/Work/WebKit/OpenSource/WebKitBuild/Debug/DerivedSources/WebKit/GeneratedWebKitSecureCoding.h:28:
In file included from /Volumes/Work/WebKit/OpenSource/Source/WebKit/Shared/Cocoa/CoreIPCTypes.h:28:
In file included from /Volumes/Work/WebKit/OpenSource/Source/WebKit/Shared/Cocoa/CoreIPCArray.h:30:
In file included from /Volumes/Work/WebKit/OpenSource/Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.h:28:
In file included from /Volumes/Work/WebKit/OpenSource/Source/WebKit/Platform/IPC/ArgumentCoders.h:30:
/Volumes/Work/WebKit/OpenSource/Source/WebKit/Platform/IPC/Decoder.h:159:40: note: in instantiation of function template specialization 'IPC::ArgumentCoder<WTF::Ref<const WebCore::FragmentedSharedBuffer>>::decode<IPC::Decoder, const WebCore::FragmentedSharedBuffer>' requested here
            std::optional<T> t { Impl::decode(*this) };
/Volumes/Work/WebKit/OpenSource/WebKitBuild/Debug/DerivedSources/WebKit/GeneratedSerializers.mm:45001:45: note: in instantiation of function template specialization 'IPC::Decoder::decode<WTF::Ref<const WebCore::FragmentedSharedBuffer>>' requested here
    auto asFragmentedSharedBuffer = decoder.decode<Ref<const WebCore::FragmentedSharedBuffer>>();
```